libs/sdk-core/src/breez_services.rs

```diff
@@ -184,11 +189,28 @@ impl BreezServices {
     ///
     /// * `bolt11` - The bolt11 invoice
     /// * `amount_sats` - The amount to pay in satoshis
-    pub async fn send_payment(&self, bolt11: String, amount_sats: Option<u64>) -> Result<()> {
+    pub async fn send_payment(
+        &self,
+        bolt11: String,
+        amount_sats: Option<u64>,
+    ) -> Result<LnPaymentDetails> {
         self.start_node().await?;
-        self.node_api.send_payment(bolt11, amount_sats).await?;
+        let payment = self.node_api.send_payment(bolt11, amount_sats).await?;
         self.sync().await?;
-        Ok(())
+
+        // Lookup the payment in our DB and return the associated payment details
+        match self
+            .persister
+            .get_payment_by_hash(&hex::encode(payment.payment_hash))?
```

Comment:
@ok300 did you consider returning the payment structure directly from the NodeAPI::send_payment?
The conversion method from greenlight payment to our structure exists there and can be used instead of taking it from the persister post payment.

Reply:
That's a very good point.

I tinkered with different variants of this and managed to get rid of the persister lookup, instead return the Payment structure directly.

Unfortunately I didn't manage to get rid of the `test_utils::CLOUD_PAYMENTS` global, because I need to tell the `MockNodeAPI` what payments it should "know about" _before_ paying the invoice with `send_payment`.

One concern I have is that this currently only works "on-the-fly", meaning the AES payload is decrypted when the LNURL workflow returns. If the user wants to later view the LNURL-pay AES "outcome", it's currently not possible AFAIK, or am I missing something? Do we explicitly store the success action fields?

Reply:
This approach looks a lot better IMO!
Persisting the success action is a good point. We were thinking to avoid that until we have a cloud backup since a restored node won't be able to get it but maybe it is a good idea to decouple these two issues and implement the persistence as you suggested so at least it works for a new node.
@ok300  How about adding an optional field `LnUrlSuccessAction` in the `LnPaymentDetails`?

Reply:
I tried different approaches, the closest to a working solution is #89. Please see the open question there.